js/ui/map.jsx

```diff
@@ -0,0 +1,63 @@
+import React, { useState } from 'react';
+import { StyleSheet, View, Image } from 'react-native';
+import MapLibreGL from '@maplibre/maplibre-react-native';
+import { getHost, getPublishableKey } from '../helpers';
+
+const DEFAULT_STYLE = 'radar-default-v1';
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+  },
+  map: {
+    flex: 1,
+  },
+  logo: {
+    position: 'absolute',
+    bottom: -10,
+    left: 5,
+    width: 50,
+    height: 50,
+    resizeMode: 'contain',
+  },
+});
+
+const createStyleURL = async (style = DEFAULT_STYLE) => {
+  const host = await getHost();
+  const publishableKey = await getPublishableKey();
+  return `${host}/maps/styles/${style}?publishableKey=${publishableKey}`;
+};
+
+const RadarMap = ({ mapOptions, children }) => {
+  const [styleURL, setStyleURL] = useState(null);
+  createStyleURL(mapOptions?.mapStyle || DEFAULT_STYLE).then((result) => {
+    setStyleURL(result);
+  });
```

Comment:
Does this not trigger an infinite loop? It looks like this function is called during each render (and setting the state)

Reply:
Good catch. Fixed: https://github.com/radarlabs/react-native-radar/pull/232/commits/d7b78b6a71c9351bc830e03b619c88dbc37f54ac